pysteps/feature/tstorm.py

```diff
@@ -0,0 +1,320 @@
+        dBZ with a resolution of 1 km. Nan values are ignored.
+    minref : float, optional
+        Lower threshold for object detection. Lower values will be set to NaN.
+        The default is 35 dBZ.
+    maxref : float, optional
+        Upper threshold for object detection. Higher values will be set to this value.
+        The default is 48 dBZ.
+    mindiff : float, optional
+        Minimal difference between two identified maxima within same area to split area
+        into two objects. The default is 6 dBZ.
+    minsize : float, optional
+        Minimal area for possible detected object. The default is 50 pixels.
+    minmax : float, optional
+        Minimum value of maximum in identified objects. Objects with a maximum lower
+        than this will be discarded. The default is 41 dBZ.
+    mindis : float, optional
+        Minimum distance between two maxima of identified objects. Objects with a
+        smaller distance will be merged. The default is 10 km.
+    dyn_thresh: bool, optional
+        Set to True to activate dynamic lower threshold. Restricts contours to more
+        meaningful area. The default is False.
+    output_feat: bool, optional
+        Set to True to return only the cell coordinates.
+    time : string, optional
+        Date and time as string. Used to label time in the resulting dataframe.
+        The default is '000000000'.
+
+    Returns
+    -------
+    cells_id : pandas dataframe
+        Pandas dataframe containing all detected cells and their respective properties
+        corresponding to the input image.
+    labels : array-like
+        Array of shape (m,n), grid of labelled cells.
+    """
+    if not SKIMAGE_IMPORTED:
+        raise MissingOptionalDependency(
+            "skimage is required for thunderstorm DATing " "but it is not installed"
+        )
+    if not PANDAS_IMPORTED:
+        raise MissingOptionalDependency(
+            "pandas is required for thunderstorm DATing " "but it is not installed"
+        )
+    filt_image = np.zeros(input_image.shape)
+    filt_image[input_image > minref] = input_image[input_image > minref]
+    filt_image[input_image > maxref] = maxref
+    max_image = np.zeros(filt_image.shape)
+    max_image[filt_image == maxref] = 1
+    labels, n_groups = ndi.label(max_image)
+    for n in range(1, n_groups + 1):
+        indx, indy = np.where(labels == n)
+        if len(indx) > 3:
+            max_image[indx[0], indy[0]] = 2
+    filt_image[max_image == 2] = maxref + 1
+    binary = np.zeros(filt_image.shape)
+    binary[filt_image > 0] = 1
+    labels, n_groups = ndi.label(binary)
+    for n in range(1, n_groups + 1):
+        ind = np.where(labels == n)
+        size = len(ind[0])
+        maxval = np.nanmax(input_image[ind])
+        if size < minsize:
+            binary[labels == n] = 0
+            labels[labels == n] = 0
+        if maxval < minmax:
+            binary[labels == n] = 0
+            labels[labels == n] = 0
```

Comment:
Perhaps good to add some comments in these lines.

Reply:
I added descriptions to lines 137 and 140 to describe the steps being done here.